tests/tracer/test_tracer.py

```diff
@@ -1706,3 +1708,30 @@ def test_tracer_api_version():
 
     t.configure(api_version="v0.4")
     assert isinstance(t.writer._encoder, MsgpackEncoderV03)
+
+
+@pytest.mark.parametrize("enabled", [True, False])
+def test_tracer_memory_leak(enabled):
+    spans = weakref.WeakSet()
+
+    # Filter to ensure we don't send the traces to the writer
+    class DropAllFilter:
+        def process_trace(self, trace):
+            return None
+
+    t = Tracer()
+    t.enabled = enabled
+    t.configure(settings={"FILTERS": [DropAllFilter()]})
```

Comment:
Would it make sense to `assert bool(spans) == enabled` and do away with the filter?

Reply:
This filter is just to ensure we don't actually try to write these spans to a trace agent / skip the writer entirely.

a docstring and maybe a different test name is a good idea, since we are basically testing the tracer + span processors for memory leaks.

although this doesn't help us if there is a leak in the writer ?


also it should always be `assert bool(spans) is False` since we want to make sure whether the tracer is enabled or not we don't hold onto span references.

Reply:
> also it should always be `assert bool(spans) is False` since we want to make sure whether the tracer is enabled or not we don't hold onto span references.

In my mind the `True` case would hold span references into a trace. If we're trying to avoid writing to a writer, can't we just use the `DummyWriter` and try to keep the tracer processing as close as possible to what the typical user would be running?

Reply:
> If we're trying to avoid writing to a writer, can't we just use the DummyWriter and try to keep the tracer processing as close as possible to what the typical user would be running?

We could, but the `DummyWriter` would hold references to them, so we'd have to clear it out.

That should be ok too.